File: 2020/02/06/cube.java
class Solution {
    public boolean isValid(String s) {
        boolean result = false;
        int cnt = 0;
        String tempStr = s;
        String compareStr = "";
            
        if(s.length()%2 == 1){
            result = false;
        } else {
            while(true){
                tempStr = tempStr.replaceAll("\\(\\)", "");
                tempStr = tempStr.replaceAll("\\{\\}", "");
                tempStr = tempStr.replaceAll("\\[\\]", "");
                
                if(compareStr.equals(tempStr)){
                    break;
                }
                
                compareStr = tempStr;
                cnt++;
            }
            
            if("".equals(tempStr)){
                result = true;
            } else {
                result = false;
            }
        }
        return result;
    }
}

File: 2020/02/06/cube.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    var tempStr = s;
    var compStr = "";
    while(true){
        tempStr = tempStr.replace("\(\)", "");
        tempStr = tempStr.replace("\{\}", "");
        tempStr = tempStr.replace("\[\]", "");
        
        if(compStr === tempStr){
            break;
        }
        compStr = tempStr;
    }
    
    if("" === tempStr){
        return true;
    }
    
    return false;
};
